test/functional/android/App.spec.js

```diff
@@ -101,7 +115,50 @@ describe("App", () => {
           .text().should.become('ON')
           .elementByXPath('//android.view.ViewGroup[1]/android.widget.TextView[1]')
           .text().should.become('decrypt');
+         
+      });
+
+      it('Encrypts text', function(){
+        
+        return driver
+          .elementByXPath('//android.widget.Switch[1]').tap()
+          .elementByXPath('//android.widget.Switch[1]')
+          .text().should.become('OFF')
+          .elementByXPath('//android.widget.EditText[1]')
+          .clear()
+          .elementByXPath('//android.widget.EditText[1]')
+          .type('This is my important message')
+          .elementByXPath('//android.view.ViewGroup[1]/android.widget.TextView[1]')
+          .tap()
+          .elementByXPath('//android.widget.EditText[1]')
+          .text()
+          .then( text => {
+            let encryptedJSON = JSON.parse(text);
+            encryptedJSON.iter.should.eql(1000);
+            encryptedJSON.ks.should.eql(128);
+            encryptedJSON.ts.should.eql(64);
+            encryptedJSON.mode.should.eql('ccm');
+            encryptedJSON.cipher.should.eql('aes');
+            encryptedJSON.ct.should.exist;
+            encryptedJSON.iv.should.exist;
+          })
+          .elementByXPath('//android.widget.EditText[1]')
+          .clear();
+
+      });
+
+      it('Decrypts text that is pasted ("typed") in', () => { 
+        return driver
+          .elementByXPath('//android.widget.EditText[1]')
+          .clear()
+          .elementByXPath('//android.widget.EditText[1]')
+          .type(thisIsTheModernWorldCT)
+          .elementByXPath('//android.view.ViewGroup[1]/android.widget.TextView[1]')
+          .tap()
+          .elementByXPath('//android.widget.EditText[1]')
+          .text().should.become('this is the modern world');
       });
+      
```

Comment:
Not crucial for right now (and not needed before merging), but might be nice to write a "round trip" test here? Ie::

* set box to encrypt mode
* type "this is the modern world"
* tap "encrypt"
* store outputted cyphertext
* tap "decrypt"
* paste cyphertext into textbox
* tap "decrypt"
* outputted clear text should equal "this is the modern world"